x-pack/plugins/security_solution/scripts/extract_tactics_techniques_mitre.js

```diff
@@ -67,29 +67,110 @@ const getSubtechniquesOptions = (subtechniques) =>
 }`.replace(/(\r\n|\n|\r)/gm, ' ')
   );
 
-const getIdReference = (references) =>
-  references.reduce(
-    (obj, extRef) => {
-      if (extRef.source_name === 'mitre-attack') {
-        return {
-          id: extRef.external_id,
-          reference: extRef.url,
-        };
+const getIdReference = (references) => {
+  const ref = references.find((r) => r.source_name === 'mitre-attack');
+  if (ref != null) {
+    return {
+      id: ref.external_id,
+      reference: ref.url,
+    };
+  } else {
+    return { id: '', reference: '' };
+  }
+};
+
+const extractTacticsData = (mitreData) => {
+  const tactics = mitreData
+    .filter((obj) => obj.type === 'x-mitre-tactic')
+    .reduce((acc, item) => {
+      const { id, reference } = getIdReference(item.external_references);
+
+      return [
+        ...acc,
+        {
+          displayName: item.name,
+          shortName: item.x_mitre_shortname,
+          id,
+          reference,
+        },
+      ];
+    }, []);
+
+  return sortBy(tactics, 'displayName');
+};
+
+const normalizeTacticsData = (tacticsData) => {
+  return tacticsData.map((data) => {
+    const { displayName, id, reference } = data;
+    return { name: displayName, id, reference };
+  });
+};
+
+const extractTechniques = (mitreData) => {
+  const techniques = mitreData
+    .filter(
+      (obj) =>
+        obj.type === 'attack-pattern' &&
+        (obj.x_mitre_is_subtechnique === false || obj.x_mitre_is_subtechnique === undefined)
```

Comment:
This script didn't work out of the box and I have to add this condition to it:

`|| obj.x_mitre_is_subtechnique === undefined`

because it looks like the official metadata has been changed since we used the script last time, and now a lot of techniques don't have `x_mitre_is_subtechnique` field in them.